lib/app/features/notifications/push_notifications.dart

```diff
@@ -0,0 +1,168 @@
+import 'dart:io';
+
+import 'package:appwrite/models.dart' hide Log;
+import 'package:firebase_messaging/firebase_messaging.dart';
+import 'package:flutter/foundation.dart';
+import 'package:gordon_ferguson_app/app/config/logger.dart';
+import 'package:gordon_ferguson_app/app/config/router.dart';
+import 'package:gordon_ferguson_app/app/features/posts/data/wordpress_client.dart';
+import 'package:gordon_ferguson_app/app/features/posts/view/screens/post_detail_view.dart';
+import 'package:gordon_ferguson_app/app/features/settings/data/shared_preferences.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:appwrite/appwrite.dart';
+
+part 'push_notifications.g.dart';
+
+@Riverpod(keepAlive: true)
+class PushNotifications extends _$PushNotifications {
+  static const tokenKey = "notificationToken";
+  static const deviceIdKey = "deviceId";
+  static const projectId = '67424e230025790edf98';
+  static const databaseId = "gordon_ferguson_teachings";
+  static const collectionId = "tokens";
+  static const topicId = "new_post";
+
+  late final FirebaseMessaging firebaseMessaging;
+  late final SharedPreferences sharedPreferences;
```

Comment:
fyi https://android-developers.googleblog.com/2020/09/prefer-storing-data-with-jetpack.html
https://pub.dev/packages/shared_preferences#migrating-from-sharedpreferences-to-sharedpreferencesasyncwithcache
you could wait on this one if you want to provide the users value sooner

Reply:
Dang that's unfortunate

Reply:
It's unfortunate because SharedPreferences is being completely deprecated, meaning all data for android will be deleted if it is not migrated. 
Luckily for us I guess we are not storing much important information with shared prefs across both apps